tests/features/environment.py

```diff
@@ -687,12 +688,16 @@ def before_all(context):
 
 def after_all(context):
     with session_ctx() as session:
-        for table in context.objects.values():
-            for record in table:
-                obj = session.get(record.__class__, record.id)
-                if obj:
-                    session.delete(obj)
+        for table in reversed(Base.metadata.sorted_tables):
+            if table.name in ("alembic_version", "parameter"):
+                continue
+            elif table.name.startswith("lexicon"):
+                continue
+            elif table.name == "sample":
+                continue
```

Comment:
if this was implemented because of a foreign key/non-nullable constraint from `sample` to `field_activity` that is going to be addressed in the DB models. @ksmuczynski and I talked about how we need to add `cascade="DELETE"` to the foreign key definition. I'm going to open a PR with this. 

So, if that is the case, I think that handling samples separately from other tables can be removed from this PR.